libbpf-rs/src/map.rs

```diff
@@ -483,6 +507,32 @@ pub trait MapCore: Debug + AsFd + private::Sealed {
         lookup_raw(self, key, flags, out_size)
     }
 
+    /// Looks up a map value into a pre-allocated buffer, avoiding allocation.
+    ///
+    /// This method provides a zero-allocation alternative to [`Self::lookup()`].
+    ///
+    /// `key` must have exactly [`Self::key_size()`] elements.
+    /// `value` must have exactly [`Self::value_size()`] elements.
+    ///
+    /// Returns `Ok(true)` if the key was found and the buffer was filled,
+    /// `Ok(false)` if the key was not found, or an error.
+    ///
+    /// If the map is one of the per-cpu data structures, this function cannot be used.
+    /// If the map is of type `bloom_filter`, this function cannot be used.
+    fn lookup_into(&self, key: &[u8], value: &mut [u8], flags: MapFlags) -> Result<bool> {
+        check_not_bloom_or_percpu(self)?;
+
+        if value.len() != self.value_size() as usize {
```

Comment:
size check is done here and not in `lookup_raw_ptr` since `lookup_raw` does not require it